meta-mel/conf/local.conf.append

```diff
@@ -14,3 +14,7 @@ IMAGE_FEATURES_DISABLED_PRODUCTION ?= "${IMAGE_FEATURES_DEVELOPMENT} ssh-server-
 # build gets its own identifier, so is self-contained already.
 # Default: ${SDKMACHINE}-${IMAGE_BASENAME}-${MACHINE}-${ADE_VERSION}
 #ADE_IDENTIFIER ?= "${SDKMACHINE}-${IMAGE_BASENAME}-${MACHINE}-${ADE_VERSION}.customized"
+SYSTEMD_BINDS = "\
+/var/volatile/systemd /etc/systemd\n\
+ "
+VOLATILE_BINDS_append = "${@bb.utils.contains('EXTRA_IMAGE_FEATURES', 'read-only-rootfs', d.getVar('SYSTEMD_BINDS', True), '', d)}"
```

Comment:
This is still both not appropriate and not necessary. As I said in the previous review, volatile-binds is only installed on read only images, so this conditional is pointless, and it needs to be conditional on the systemd distro feature.

Reply:
As I said on IRC: it's not appropriate to muck with image features globally. It's completely valid to have one image be read-only-rootfs and another image non-read-only-rootfs in a single bitbake build, which would not work properly the way you're doing things. Absolute worst case we can use a rootfs postprocessing hook to alter the configuration after rootfs construction, based on the image feature, the way a number of other read only rootfs tweaks are made already.

If you grep oe-core for read-only-rootfs, you'll see the only time anything uses it is in image context. Recipe contents never change based on image features.

The difficulty with trying to do anything like this via a rootfs hook is that volatile-binds mounts are set up via individual systemd services, based on a template in volatile-binds SRC_URI, which is something  we won't easily get to in an image. Given this, I'd suggest arranging (likely via bbappend) to unconditionally add the needed entry to VOLATILE_BINDS, but disable the service generated for it by default, then enable it in a rootfs postprocessing hook only when read-only-rootfs and the systemd distro feature are both enabled.